src/main/java/com/growup/pms/user/service/UserService.java

```diff
@@ -69,7 +72,23 @@ public UserDownloadCommand imageDownload(Long userId) {
 
         return new UserDownloadCommand(user.getProfile().getImageName(), storageService.getFileResource(path));
     }
-  
+
+    @Transactional
+    public void updatePassword(Long userId, UserPasswordUpdateCommand command) {
+        User user = validatePassword(userId, command.password());
+
+        user.changePassword(passwordEncoder, command.passwordNew());
+    }
+
+    private User validatePassword(Long userId, String password) {
+        User user = userRepository.findByIdOrThrow(userId);
+
+        if (!passwordEncoder.matches(password, user.getPassword())) {
+            throw new BusinessException(ErrorCode.USER_NOT_FOUND);
+        }
+        return user;
+    }
```

Comment:
`validatePassword`가 어떤 대상을 검증하는 동작과 동시에 쿼리의 성격도 가지고 있는 것 같습니다! 그리고 해당 메서드에서 필요한 필드만 넘기면 좋을 것 같아요 :thinking: 

```suggestion
    @Transactional
    public void updatePassword(Long userId, UserPasswordUpdateCommand command) {
        User currentUser = userRepository.findByIdOrThrow(userId);

        validateCurrentPassword(command.password(), currentUser.getPassword());

        currentUser.changePassword(passwordEncoder, command.passwordNew());
    }

    private void validateCurrentPassword(String inputPassword, String storedPassword) {
        if (!passwordEncoder.matches(inputPassword, storedPassword)) {
            throw new BusinessException(ErrorCode.INVALID_PASSWORD);
        }
    }
```